core/class/luna.class.php

```diff
@@ -869,8 +869,88 @@ public static function recuperationConfigModem() {
 
   /* ------ FIN 4G ----- */
 
+  /* ----- DEBUT SMS  ----- */
+
+  public static function listGlobalSMS(){
+    $list = self::listSMS();
+    $return = [];
+    foreach ($list as $sms) {
+      $sms = self::readSMS($sms);
+      $return[] = $sms;
+    }
+    return $return;
+  }
+
+  public static function listSMS() {
+    $list = shell_exec('sudo mmcli -m 0 --messaging-list-sms');
```

Comment:
Consider error handling for shell_exec calls. If these calls fail for any reason, the code should handle these failures gracefully. [important]